providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -142,7 +146,13 @@ static int rsa_decrypt(void *vprsactx, unsigned char *out, size_t *outlen,
         return 1;
     }
 
-    if (prsactx->pad_mode == RSA_PKCS1_OAEP_PADDING) {
+    if (outsize < len) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_BAD_LENGTH);
+        return 0;
+    }
+
+    if (prsactx->pad_mode == RSA_PKCS1_OAEP_PADDING
+            || prsactx->pad_mode == RSA_PKCS1_WITH_TLS_PADDING) {
         int rsasize = RSA_size(prsactx->rsa);
```

Comment:
rsasize == len

Reply:
I removed rsasize and just used len instead.